presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetPageReader.java

```diff
@@ -74,14 +75,18 @@ public ParquetDataPage readPage()
                 if (!dataPageV2.isCompressed()) {
                     return dataPageV2;
                 }
+                int uncompressedSize = Ints.checkedCast(
```

Comment:
this is how [ParquetFileReader](https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java#L875) sets the uncompressed size for v2 data pages.